### PR TITLE
Prepare regrid pipeline to be used in Prefect

### PR DESCRIPTION
**Pull request by Joshdpaul**
This PR is a companion PR to [PR #22 ](https://github.com/ua-snap/prefect/pull/22)for the `regridding` [branch of the prefect repo](https://github.com/ua-snap/prefect/tree/regridding/regridding).

Some changes were made to the CMIP6 regridding pipeline in order to make it work via Prefect.:

- Instead of using `config.py` and environmental variables, the various directories needed for the pipeline are just constructed from the `{scratch_directory}` parameter in the `regrid_cmip6.regrid_cmip6()` function. (This is similar to what we did in the inidicators flow.) As a result, more arguments have to be passed to `slurm.py` and `regrid.py` for all the directories since they are not readily available from `config.py`.
- A new `run_generate_batch_files.py` script was needed to pass parameters to `generate_batch_files.py`, create the sbatch head and slurm file, and submit a the slurm job for that process.
- The step of submitting the slurm jobs was moved out of the `regrid_cmip6.ipynb` notebook and into the main block of `slurm.py`
- Minor changes to the `no_clobber` parameter. It has to be a string instead of a command line "action". (Not sure of the nomenclature here, but from Prefect we need to provide a string "true"/"false" as the input instead of the inclusion of ```--no_clobber``` as a flag in the original command line version of `regrid.py`. 

**TO TEST:**

- Clone the `regridding` [branch of the prefect repo](https://github.com/ua-snap/prefect/tree/regridding/regridding).
- Review the code changes to `slurm.py` and `regrid.py` and the corresponding arguments passed to tasks in the prefect flow. 
- Review the code in the new `run_generate_batch_files.py` script. Should this be rolled into the `generate_batch_files.py` or is it OK to have that script kind of "wrapped" by this new one?
- Complete the companion prefect PR to run the regridding pipeline on a subset of files, and confirm it works as expected.

### CODE REVIEW

**Review by BobTorgerson — APPROVED** [review]
These changes all worked when paired with the new Prefect flow for regridding and I was able to generate the regridded files as the snapdata user. 

I don't have an issue with the wrapper script (run_generate_batch_files.py) being used as a means to provide data from Prefect to the generate_batch_files functionality. 

I approve these changes!

**Review by kyleredilla — CHANGES_REQUESTED** [review]
Ahhh I think I forgot to hit submit review on this yesterday, sorry! 

Just a couple of suggestions, but also works great as is so I approve regardless.

File: regridding/slurm.py [hunk 1]
```diff
@@ -59,7 +56,7 @@ def write_sbatch_regrid(
         regrid_dir (pathlib.PosixPath): directory to write the regridded data to
         regrid_batch_fp (path_like): path to the batch file containing paths of CMIP6 files to regrid
         dst_fp (path_like): path to file being used as template / reference for destination grid
-        no_clobber (bool): do not overwrite regridded files if they exist in regrid_dir
```
[the remark this answers is not in the document]
Comment (Joshdpaul) [reply regridding/slurm.py:62]: @kyleredilla so in order to pass an argument from the Prefect flow, it has to be something that can be stored in a JSON dict. And then the argument (in this case a string "true"/"false") has to be passed thru all the functions until it ends up being evaluated by `slurm.py`. I couldn't really figure out how to reconcile this with `action` type arguments that are simply included/not included from the command line script when writing the sbatch file. I tried this many different ways and got errors with this argument every time, so I just changed it to a string.

If you follow the argument from the Prefect flow through the functions, I think you will see what I am getting at.

**Comment by BobTorgerson** [comment]
We've reviewed this code, so I'm going to go ahead and merge this into main.

**Pull request merged**